Run automated tests in Appveyor

dipinhora opened the pull request:
This pull request enables running the normal ponyc automated tests (that run for linux/osx) on every commit in Appveyor for Windows builds.

See https://ci.appveyor.com/project/dipinhora/ponyc for a build run with Appveyor with these changes included. Currently a couple of tests fail so the build fails.

This pull request also includes changes to how the Visual Studio installation location is determined by ponyc for finding libraries and executables needed to include support for systems where only the command line Visual C++ Build Tools 2015 (https://blogs.msdn.microsoft.com/vcblog/2016/03/31/announcing-the-official-release-of-the-visual-c-build-tools-2015/) are installed but a full copy of Visual Studio isn't installed. This change should have no negative impact to normal Visual Studio installs (as confirmed by the Appveyor build referenced above) but it would be ideal if other Windows users such as @kulibali, @sblessing, or others are able to confirm.

Also, I am not 100% sure that the caching functionality in Appveyor is working correctly (although in theory it should be based on my understanding of the docs) with this so that might need tweaking to get functioning properly.

kulibali commented:
I can confirm that ponyc from your fork can still build and link on my machine with a full VS2015 install.

dipinhora commented:
@kulibali, thanks for confirming. Glad to hear it still works for you.

dipinhora commented:
Seems one of the Travis OS X builds failed on one of the Process package tests. Not sure why/how that is related since the changes in this PR are isolated to Windows and Appveyor.

Also, the Appveyor tests failed as expected but it seems the first one failed due to a curl timeout instead of due to failed tests like the others. We might need to find a more reliable place to get the pcre2 sources from so that doesn't keep happening. The reason I switched to sourceforge instead of ftp.csx.cam.ac.uk is because Appveyor seemed to always timeout on the ftp from this site.

jemc commented:
Yes, I'm sure the Process package failure on Travis is unrelated.

Can we exclude the failing tests on windows using `ifdef` so that the appveyor build will (usually) pass before merging?

SeanTAllen commented:
what tests would those be @jemc?

jemc commented:
> @dipinhora wrote:
> Currently a couple of tests fail so the build fails.

From looking at the appveyor output, it looks like those are these:

> **** FAILED: 2 tests, listed below:
**** FAILED: net/TCP.expect
**** FAILED: net/TCP.writev

SeanTAllen commented:
is that windows specific or flakey test?

jemc commented:
From @dipinhora's description it sounds like they fail reliably on appveyor.

The specific failure output I see on the appveyor output is unique, in that I haven't seen it on Travis before - even though some of TCP tests sometimes fail on Travis, they fail in a different way, with different "action" failure output.

dipinhora commented:
The tests only fail on Windows and they fail consistently. Not sure of the root cause but it seems to be a TCP connection issue in the runtime C code. My limited efforts at figuring it out via print statements seem to indicate that the call to `getaddrinfo` in https://github.com/ponylang/ponyc/blob/master/src/libponyrt/lang/socket.c#L116 is failing and the function is always returning `NULL` on Windows but I don't quite know why or how to fix it.

I'll exclude the tests via an ifdef as per @jemc for now.

SeanTAllen commented:
@jemc @dipinhora we can close this now yes?

dipinhora commented:
Yes, I made the change to exclude the failing tests via ifdefs and I also removed an unnecessary dependency and changed it to pull the source for pcre2 via an svn checkout instead of a sourceforge download.
